core/remotes/docker/fetcher.go

```diff
@@ -35,10 +37,168 @@ import (
 	"github.com/klauspost/compress/zstd"
 	digest "github.com/opencontainers/go-digest"
 	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"golang.org/x/sync/semaphore"
 )
 
+type BufferPool struct {
+	sync.Mutex
+	sem    *semaphore.Weighted
+	pool   []bytes.Buffer
+	bufCap int64
+}
```

Comment:
can we use `sync.Pool`(https://pkg.go.dev/sync#Pool) directly? There are some examples of this in containerd code.

Reply:
good point, updated !